src/Symfony/Component/Form/Extension/Core/Type/ChoiceType.php

```diff
@@ -244,6 +244,18 @@ public function finishView(FormView $view, FormInterface $form, array $options)
                 $childView->vars['full_name'] = $childName;
             }
         }
+
+        // We have to override parent {@link FormType} condition of the empty_view option here,
+        // even if there is no choices the field view may not be empty because:
+        //
+        //  - a select input is not "compound" but should be considered as any empty HTML tag,
+        //
+        //  - when expanded it could hold a placeholder which has been counted as a children,
+        //    we should ignore it in any case since its role is to be a "null" alternative to
+        //    choices, not a choice itself. Think of "Choose something" with no choices.
+        if (array() === $view->vars['choices'] && array() === $view->vars['preferred_choices']) {
```

Comment:
@webmozart Because the view var `empty_view` is set [here](https://github.com/symfony/symfony/pull/17609/files/94f2a70c79494a4b248acf4cfb545757886dd059#diff-22a03074077455f3d1128486ef5d9851R118) but is not compatible with the only choice type.
I just thought we should resolve the state of this view var from within this method to avoid holding too much logic in the view.

I don't know how BC could be broken with such changes in form themes. the fact that I modify several test with the current implementation is not really reassuring. What do you think ?